apis/python/src/tiledbsoma/io/conversions.py

```diff
@@ -46,7 +46,7 @@ def to_tiledb_supported_array_type(name: str, x: _MT) -> _MT:
     E.g., float16 -> float32
     """
     if isinstance(x, (np.ndarray, sp.spmatrix)) or not isinstance(
-        x, pd.CategoricalDtype
+        x.dtype, pd.CategoricalDtype
```

Comment:
I think this `if` condition was trivially evaluating as `True`, previously ([last commit](https://github.com/single-cell-data/TileDB-SOMA/commit/616823f4697c08db3043319cc68642551aaa44f5#diff-14e9163b10b9e187b5e590445e41f83c891a991bc670b9bda9cf194217532fc4L49-R50)).